docs/API.md

```diff
@@ -10,169 +10,166 @@ redirect_from:
 // service.js
 module.exports = async function() {
     // This service needs to be registered for the module to work
-	// but it will be used later in the "Receiving Events" section
+    // but it will be used later in the "Receiving Events" section
 }
 ```
 
 ## Controlling the Player
 
-### Initializing a Track
-Using the newly created player, you can add a media file from the network or from the app bundle.
+### Adding Tracks to the Playback Queue
+You can add a track to the player using a url or by requiring a file in the app bundle or on the file system.
 
-First of all, you need to create a [Track Structure](https://react-native-kit.github.io/react-native-track-player/documentation/#track-structure) to provide all the information for the player
+First of all, you need to create a [track object](https://react-native-kit.github.io/react-native-track-player/documentation/#track-object), which is a plain javascript object with a number of properties describing the track. Then [add](https://react-native-kit.github.io/react-native-track-player/documentation/#addtracks-insertbeforeid) the track to the queue:
 
 ```javascript
-var track = {
-    id: 'unique track id', // Must be a string, required
-    
-    url: 'http://example.com/avaritia.mp3', // Load media from the network
-    url: require('./avaritia.ogg'), // Load media from the app bundle
-    url: 'file:///storage/sdcard0/Music/avaritia.wav' // Load media from the file system 
-
+const track1 = {
+    id: 'avaritia', // Must be a string, required
+    url: 'http://example.com/avaritia.mp3', // Load media through the network
     title: 'Avaritia',
     artist: 'deadmau5',
     album: 'while(1<2)',
     genre: 'Progressive House, Electro House',
     date: '2014-05-20T07:00:00+00:00', // RFC 3339
-    
-    artwork: 'http://example.com/avaritia.png', // Load artwork from the network
-    artwork: require('./avaritia.jpg'), // Load artwork from the app bundle
-    artwork: 'file:///storage/sdcard0/Downloads/artwork.png' // Load artwork from the file system
+    artwork: 'http://example.com/avaritia.png' // Load artwork from the network
 };
-```
-[Check the documentation](https://react-native-kit.github.io/react-native-track-player/documentation/#track-structure) for more information about the properties you can have in a track structure.
 
-After creating it, you can [add](https://react-native-kit.github.io/react-native-track-player/documentation/#addtracks-insertbeforeid) it to the queue:
+const track2 = {
+    id: 'coelacanth',
+    url: require('./avaritia.ogg'), // Load media from the app bundle
+    title: 'Coelacanth I',
+    artist: 'deadmau5',
+    artwork: require('./avaritia.jpg') // Load artwork from the app bundle
+};
 
-```javascript
-// Adding to the queue
-// You can provide multiple items in one single call
-TrackPlayer.add([track, track2]).then(function() {
-    // The tracks were added
-});
+// Add the tracks to the queue:
+await TrackPlayer.add([track1, track2]);
 ```
 
```

Comment:
Yes - do we also need the track durations in there?

Reply:
Yeah, there's recurring issues that reports that `getDuration()` returns the wrong value. The answer is that the module streams the track, it doesn't load the full track into memory before playing it, so the duration is pretty much a prediction. Because of that, all streaming services store the duration in their database in order to not have to rely on the duration calculated by the player. There is a `duration` parameter in the track object that will make `getDuration()` return it.

It would be really helpful if you could add that explanation somewhere

Reply:
Okay. I think the `getDuration()` docs cover your explanation: https://react-native-kit.github.io/react-native-track-player/documentation/#getduration but I will see if I can reword it a bit